src/RhythmRunner.cpp

```diff
@@ -35,38 +28,76 @@ int main(int argc, char** argv) {
 
   std::shared_ptr<Level> level = levelGenerator->generateLevel();
 
-  std::shared_ptr<GameState> gameState =
-      std::make_shared<GameState>(level, camera, player);
+  std::shared_ptr<GameState> game_state = std::make_shared<GameState>(
+      level, std::make_shared<GameCamera>(), std::make_shared<Player>());
+
+  std::shared_ptr<GameRenderer> renderer = std::make_shared<GameRenderer>();
+  renderer->Init(RESOURCE_DIR, game_state);
+
   std::shared_ptr<GameUpdater> updater = std::make_shared<GameUpdater>();
+  updater->Init(game_state);
 
-  renderer->Init(RESOURCE_DIR, gameState);
-  // fix ur time step
-  double clock = glfwGetTime();
-#ifdef DEBUG
-  float elapsed[25] = {1};
-  int pos = 0;
-#endif
-  while (!glfwWindowShouldClose(renderer->GetWindow())) {
-    double nextTime = glfwGetTime();
-    if (nextTime - clock > SEC_PER_FRAME) {
-      updater->Update(gameState);
-      renderer->Render(gameState);
-
-#ifdef DEBUG
-      elapsed[pos++] = glfwGetTime() - clock;
-      pos %= 25;
-      float tot = 0;
-      for (int i = 0; i < 25; i++)
-        tot += elapsed[i];
+  double initial_glfw_time = glfwGetTime();
+  // get one update in before starting rendering?
+  // TODO(jarhar): if this isnt necessary to set up the game it should be
+  // removed. Update() was originally called before Render()
+  updater->Update(game_state);
+
+  while (!renderer->WindowShouldClose()) {
+    // Render first, then catch up in Update()s
+    // TODO(jarhar): Does this order make more sense than the opposite?
+    renderer->Render(game_state);
```

Comment:
I think it makes sense to render first and then update since the player will never see the initial state when the game starts if the game updates first. Not that it makes a huge difference since that is only one frame.